inc/document_item.class.php

```diff
@@ -755,54 +755,9 @@ static function showListForItem(CommonDBTM $item, $withtemplate = 0, $options =
          $linkparam = "&amp;tickets_id=".$item->fields['id'];
       }
 
-      $criteria = [
-         'SELECT'    => [
-            'glpi_documents_items.id AS assocID',
-            'glpi_documents_items.date_creation AS assocdate',
-            'glpi_entities.id AS entityID',
-            'glpi_entities.completename AS entity',
-            'glpi_documentcategories.completename AS headings',
-            'glpi_documents.*'
-         ],
-         'FROM'      => 'glpi_documents_items',
-         'LEFT JOIN' => [
-            'glpi_documents'  => [
-               'ON' => [
-                  'glpi_documents_items'  => 'documents_id',
-                  'glpi_documents'        => 'id'
-               ]
-            ],
-            'glpi_entities'   => [
-               'ON' => [
-                  'glpi_documents'  => 'entities_id',
-                  'glpi_entities'   => 'id'
-               ]
-            ],
-            'glpi_documentcategories'  => [
-               'ON' => [
-                  'glpi_documentcategories'  => 'id',
-                  'glpi_documents'           => 'documentcategories_id'
-               ]
-            ]
-         ],
-         'WHERE'     => [
-            'glpi_documents_items.items_id'  => $item->getID(),
-            'glpi_documents_items.itemtype'  => $item->getType()
-         ],
-         'ORDERBY'   => [
-            "$sort $order"
-         ]
-      ];
-
-      if (Session::getLoginUserID()) {
-         $criteria['WHERE'] = $criteria['WHERE'] + getEntitiesRestrictCriteria('glpi_documents', '', '', true);
-      } else {
-         // Anonymous access from FAQ
-         $criteria['WHERE']['glpi_documents.entities_id'] = 0;
-      }
```

Comment:
the above request is also needed for  the PR, then I isolated it in a dedicated method for factorization need.